evm/src/keccak_sponge/keccak_sponge_stark.rs

```diff
@@ -361,40 +361,242 @@ impl<F: RichField + Extendable<D>, const D: usize> Stark<F, D> for KeccakSpongeS
     fn eval_packed_generic<FE, P, const D2: usize>(
         &self,
         vars: StarkEvaluationVars<FE, P, { Self::COLUMNS }>,
-        _yield_constr: &mut ConstraintConsumer<P>,
+        yield_constr: &mut ConstraintConsumer<P>,
     ) where
         FE: FieldExtension<D2, BaseField = F>,
         P: PackedField<Scalar = FE>,
     {
-        let _local_values: &KeccakSpongeColumnsView<P> = vars.local_values.borrow();
+        let local_values: &KeccakSpongeColumnsView<P> = vars.local_values.borrow();
+        let next_values: &KeccakSpongeColumnsView<P> = vars.next_values.borrow();
 
-        // TODO: Each flag (full-input block, final block or implied dummy flag) must be boolean.
-        // TODO: before_rate_bits, block_bits and is_final_input_len must contain booleans.
+        // Each flag (full-input block, final block or implied dummy flag) must be boolean.
+        let is_full_input_block = local_values.is_full_input_block;
+        yield_constr.constraint(is_full_input_block * (is_full_input_block - P::ONES));
 
-        // TODO: Sum of is_final_input_len should equal is_final_block (which will be 0 or 1).
+        let is_final_block = local_values.is_final_block;
+        yield_constr.constraint(is_final_block * (is_final_block - P::ONES));
 
-        // TODO: If this is the first row, the original sponge state should be 0 and already_absorbed_bytes = 0.
-        // TODO: If this is a final block, the next row's original sponge state should be 0 and already_absorbed_bytes = 0.
+        for &is_final_len in local_values.is_final_input_len.iter() {
+            yield_constr.constraint(is_final_len * (is_final_len - P::ONES));
+        }
+
+        // Ensure that full-input block and final block flags are not set to 1 at the same time.
+        yield_constr.constraint(is_final_block * is_full_input_block);
```

Comment:
I think this could be removed, as it would end up being a dummy row which continuity with another dummy row is checked below, but as we define dummy rows as having both flags to 0, I kept this additional check in case.